File: backend/app/schemas/generation.py
import uuid

from pydantic import BaseModel


class GenerationRequest(BaseModel):
    topic: str
    platform: str
    tone: str
    style_hints: list[str] = []
    account_id: uuid.UUID | None = None


class GeneratedContent(BaseModel):
    text: str
    image_prompt: str
    image_url: str | None
    hashtags: list[str]
    estimated_engagement: float


class BatchGenerationRequest(BaseModel):
    account_id: uuid.UUID
    topics: list[str]
    platform: str
    tone: str


class RegenerateRequest(BaseModel):
    regenerate_text: bool = True
    regenerate_image: bool = True

File: frontend/src/api/generation.ts
import client from './client'

export interface GenerationRequest {
  topic: string
  platform: string
  tone: string
  style_hints: string[]
  account_id?: string
}

export interface GeneratedContent {
  text: string
  image_prompt: string
  image_url: string | null
  hashtags: string[]
  estimated_engagement: number
}

export const generationApi = {
  generate: (data: GenerationRequest) => client.post<GeneratedContent>('/generation/generate', data),
  regenerate: (postId: string, opts: { regenerate_text: boolean; regenerate_image: boolean }) =>
    client.post<GeneratedContent>(`/generation/regenerate/${postId}`, opts),
  batch: (data: { account_id: string; topics: string[]; platform: string; tone: string }) =>
    client.post<GeneratedContent[]>('/generation/batch', data),
}
